editor.planx.uk/src/@planx/components/PlanningConstraints/Public.tsx

```diff
@@ -176,7 +178,13 @@ function Component(props: Props) {
               if (data) _constraints.push(data as GISResponse["constraints"]);
             }
 
-            const _overrides = inaccurateConstraints;
+            const hasInaccurateConstraints = inaccurateConstraints && Object.keys(inaccurateConstraints).length > 0;
+            const _overrides = hasInaccurateConstraints ? { ...priorOverrides, [props.fn]: inaccurateConstraints } : undefined;
```

Comment:
An observation / pattern I didn't really notice last time round when `inaccurateConstraints` were added to this component.

This component has transitioned from being informational only, to now capturing data much like other components. This data is generally captured and validated using formik (unlike the `useState()` hook here for `inaccurateConstraints`).

This particular function is essentially handling the "back" behaviour usually managed by `getPreviouslySubmittedData()`.

If / when we come back to handle using formik for the `OverrideEntitiesModal` instances, it likely makes sense that this state would live at this root component level to match the behaviour of other components as opposed to individual instances within each modal as I think I'd previously suggested.

Reply:
Thanks for noting this here so we have record when coming back; will leave unresolved so stays expanded :+1: